design/auth.md

```diff
@@ -0,0 +1,270 @@
+# Auth Service
+
+## Context
+
+This is a rewrite of how we do authentication and authorization in the Screwdriver world.  The purpose of this is to harden our security as well as enable us to support private repositories.  Much of what is being proposed here is based on the [JWT Spec](https://tools.ietf.org/html/draft-ietf-oauth-json-web-token-32) and Docker Hub's [implementation](https://docs.docker.com/registry/spec/auth/jwt/).
+
+## New Service (Auth)
+
+We should add a new service, called `auth`, that owns authentication and authorization.  This will be the entry point for users to authenticate via OAuth or Access Tokens as well authorize Bearer Tokens for API access.
+
+### Token Types
+
+ - *Refresh* - 12h signed JWT for authenticating a user for the Auth service.
+ - *Bearer* - 5m signed JWT for authorizing specific operations against the Screwdriver API.
+
+### Login Types
+
+Tokens will be given out for logged-in users, anonymous users, and builds.  Subject will be based on the type:
+
+ - Logged-in User - `user:<YOUR-NAME-HERE>`
+ - Anonymous User - `anon:<RANDOM-ID-HERE>`
+ - Build - `build:<BUILD-ID-HERE>`
+
+## Flow Changes
```

Comment:
![image](https://user-images.githubusercontent.com/3217917/29193479-4d01585e-7dda-11e7-8c72-06da077026b8.png)
This is rendering as so, currently